backend/src/handlers/user_handler.go

```diff
@@ -157,6 +157,17 @@ func (srv *Server) HandleCreateUser(w http.ResponseWriter, r *http.Request) {
 	if user.User.FacilityID == 0 {
 		user.User.FacilityID = srv.getFacilityID(r)
 	}
+	userNameExists := srv.Db.GetUserByUsername(user.User.Username)
+	if userNameExists != nil {
+		srv.ErrorResponse(w, http.StatusInternalServerError, "Username already exists")
```

Comment:
Can we return something like BadRequest or Forbidden here instead of a 500?

Reply:
yes, all set now.  Thanks.